### PR TITLE
Implement `Selector` (#574 part5)

### PR DESCRIPTION
**Pull request by inikulin**
In this part:
- Client function `scopeVars`
- Node snapshot methods/selectors

\cc @VasilyStrelyaev @helen-dikareva @AlexanderMoskovkin

### CODE REVIEW

**Comment by AlexanderMoskovkin** [comment]
a functional test failed

**Comment by VasilyStrelyaev** [comment]
Maybe I'm a bit unfamiliar with the terms, but I do understand what `context` means, I can understand what `environment` means (although `context` seems better), but `scope variables` as a parameter name leaves me puzzled.
Maybe it's just me.

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
What the..

**Comment by inikulin** [comment]
Wow, something wrong with Travis, same issue in docs PR: https://travis-ci.org/testcafe-build-bot/testcafe/builds/139281330

Is there something wrong with Travis or gulp itself? \cc @AlexanderMoskovkin

**Comment by inikulin** [comment]
https://travis-ci.org/testcafe-build-bot/testcafe-hammerhead/jobs/139283849 - same in hammerhead repo

**Comment by inikulin** [comment]
Yeah, it's Travis issue, can be reproduced without our build bot: https://travis-ci.org/DevExpress/generator-testcafe-reporter/jobs/139285362

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
It was a bug in gulp dependency, should be fixed already

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
FPR

File: src/errors/create-stack-filter.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+export default function createStackFilter (limit) {
+    var passedFramesCount = 0;
+
+    return function stackFilter (frame) {
+        if (passedFramesCount >= limit)
+            return false;
+
+        var filename = frame.getFileName();
+
+        // NOTE: filter node, Babel and TestCafe internals
```
Comment (VasilyStrelyaev) [inline src/errors/create-stack-filter.js]:
```
        // NOTE: filter out the internals of node, Babel and TestCafe
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by inikulin** [comment]
FPR

**Comment by helen-dikareva** [comment]
lgtm

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**